### PR TITLE
Support for slices of ints, bools and strings.

### PR DESCRIPTION
**Pull request by dselans**
Hey there - nice lib!

This PR adds support for []string, []int and []bool types and allows you to set a separator via the struct tag.

```
dselans:env dselans$ go test ./... -cover -v
=== RUN   TestParsesEnv
--- PASS: TestParsesEnv (0.00s)
=== RUN   TestEmptyVars
--- PASS: TestEmptyVars (0.00s)
=== RUN   TestPassAnInvalidPtr
--- PASS: TestPassAnInvalidPtr (0.00s)
=== RUN   TestPassReference
--- PASS: TestPassReference (0.00s)
=== RUN   TestInvalidBool
--- PASS: TestInvalidBool (0.00s)
=== RUN   TestInvalidInt
--- PASS: TestInvalidInt (0.00s)
=== RUN   TestInvalidBoolsSlice
--- PASS: TestInvalidBoolsSlice (0.00s)
=== RUN   TestParsesDefaultConfig
--- PASS: TestParsesDefaultConfig (0.00s)
=== RUN   TestParseStructWithoutEnvTag
--- PASS: TestParseStructWithoutEnvTag (0.00s)
=== RUN   TestParseStructWithInvalidFieldKind
--- PASS: TestParseStructWithInvalidFieldKind (0.00s)
=== RUN   TestUnsupportedSliceType
--- PASS: TestUnsupportedSliceType (0.00s)
=== RUN   TestBadSeparator
--- PASS: TestBadSeparator (0.00s)
=== RUN   ExampleParse
--- PASS: ExampleParse (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/dselans/env	0.008s	coverage: 100.0% of statements
?   	github.com/dselans/env/examples	[no test files]
```

P.S. Will have to update the import lines in env_test.go and the example to point to your repo!

### CODE REVIEW

**Comment by dselans** [comment]
Dunno about the CirceCI tests - lmk if something is amiss.

**Comment by caarlos0** [comment]
Hi there!

Nice patch! Will check it at home! Thanks!

File: examples/first.go [hunk 1]
```diff
@@ -3,13 +3,14 @@ package main
 import (
 	"fmt"
 
-	"github.com/caarlos0/env"
```
Comment (caarlos0) [inline examples/first.go]: why this change?

File: env_test.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 	"os"
 	"testing"
 
-	"github.com/caarlos0/env"
+	"github.com/dselans/env"
```
Comment (caarlos0) [inline env_test.go]: @dselans why this?

File: examples/first.go (hunk 1 above)
Comment (dselans) [inline examples/first.go]: I switched it while writing the patch (since I was working in my fork and tests wouldn't pass otherwise). I can push a quick update to both.

File: env_test.go (hunk 2 above)
Comment (dselans) [inline env_test.go]: Same as above:

I switched it while writing the patch (since I was working in my fork) - I can push a quick update to both.

**Comment by dselans** [comment]
@caarlos0 the ci test will (probably) fail due to this being a fork - if you check it out locally and run tests, it should work fine.

```
dselans:env dselans$ pwd
/Users/dselans/Code/go/src/github.com/dselans/env
dselans:env dselans$ go test ./... -v
=== RUN   TestParsesEnv
--- PASS: TestParsesEnv (0.00s)
=== RUN   TestEmptyVars
--- PASS: TestEmptyVars (0.00s)
=== RUN   TestPassAnInvalidPtr
--- PASS: TestPassAnInvalidPtr (0.00s)
=== RUN   TestPassReference
--- PASS: TestPassReference (0.00s)
=== RUN   TestInvalidBool
--- PASS: TestInvalidBool (0.00s)
=== RUN   TestInvalidInt
--- PASS: TestInvalidInt (0.00s)
=== RUN   TestInvalidBoolsSlice
--- PASS: TestInvalidBoolsSlice (0.00s)
=== RUN   TestParsesDefaultConfig
--- PASS: TestParsesDefaultConfig (0.00s)
=== RUN   TestParseStructWithoutEnvTag
--- PASS: TestParseStructWithoutEnvTag (0.00s)
=== RUN   TestParseStructWithInvalidFieldKind
--- PASS: TestParseStructWithInvalidFieldKind (0.00s)
=== RUN   TestUnsupportedSliceType
--- PASS: TestUnsupportedSliceType (0.00s)
=== RUN   TestBadSeparator
--- PASS: TestBadSeparator (0.00s)
=== RUN   ExampleParse
--- PASS: ExampleParse (0.00s)
PASS
ok  	github.com/dselans/env	0.012s
?   	github.com/dselans/env/examples	[no test files]
```

**Comment by caarlos0** [comment]
@dselans ok, can you also fix the houndci issues?

**Comment by dselans** [comment]
Saw your commits, was reminded about PR - fixed the houndci complaints. CircleCI will fail due to the package imports coming from your repo. Might want to add godep at some point to avoid this.

**Comment by caarlos0** [comment]
@dselans yeah, I really need to do that..

Thanks!
